GOSR2.py

```diff
@@ -0,0 +1,200 @@
+
+    suitable_seqs = list(
+        record
+        for record in SeqIO.parse(input_path, "fastq")
+        if ((gc_bounds[0] <= SeqUtils.gc_fraction(record.seq)*100 <= gc_bounds[1]) and 
+            (length_bounds[0] <= len(record) <= length_bounds[1]) and 
+            (sum(record.letter_annotations['phred_quality']) / len(record) >= quality_threshold))
+    )  
+
+    if output_filename == '':
+        output_filename = 'good_quality.fastq'
+    dir_name = 'fastq_filtrator_results'
+    if not output_filename[-6:] == '.fastq':
+        output_filename = output_filename + '.fastq'
+    if not os.path.isdir(dir_name):
+        os.mkdir(dir_name)
+    output_path = os.path.join(dir_name, output_filename)
+    if os.path.isfile(output_path):
+        raise ValueError('File with such name is exist. Please, use another name for your output file.')
+    SeqIO.write(suitable_seqs, output_path, 'fastq')
+    print('The result of fastq-filtering is saved in', output_path)
+    
+    return suitable_seqs
+
+
+class BiologicalSequence(ABC, str):
+    """
+    Abstract class for NucleicAcidSequence and AminoAcidSequence. 
+    attributes:
+        - seq (str): sequence
+    methods:
+        - check_alphabet: check whether the seq matches the right biological alphabet
+    """
+
+    def __init__(self, seq: str):
+        self.seq = seq
+
+    @abstractmethod
+    def check_alphabet(self)->bool:
+        """Check whether the seq matches the right biological alphabet."""
+        
+        return set(self.seq).issubset(self.alphabet)
+
+
+class NucleicAcidSequence(BiologicalSequence):
+    """
+    Class for DNASequence and RNASequence.
+    methods:
+        - complement: return the complemetary DNA or RNA sequence
+    """
+    
+    def __init__(self, seq: str):
+        super().__init__(seq = seq)
+
+    def complement(self):
+        """Return the complemetary DNA or RNA sequence."""
+
+        try:
+            complementary_seq = ''.join([self.comlementation[nucleotide] for nucleotide in self])
+        except AttributeError:
+            raise NotImplementedError('A DNASequence or RNASequence type object is required.')
+        
+        return type(self)(complementary_seq)
+
+    def gc_content(self):
+        """Return the gc-content of in sequence in percent."""
+        
+        return 100*(self.count('G') + self.count('C')) / len(self)
+
+
+class DNASequence(NucleicAcidSequence):
+    """
+    Class for DNA sequences.
+    attributes:
+        - alphabet (str): allowed characters in the sequence
+        - comlementation (dict): dictionary with complementarity rules
+        - transcription (dict): dictionary with transcription rules
+    methods:
+        - transcribe: return the transcribed DNA
+    """
+    
+    def __init__(self, seq: str):
+        super().__init__(seq = seq)
+        
+        self.alphabet = {'A', 'T', 'G', 'C', 'a', 't', 'g', 'c'}
+        self.comlementation = {

```

Comment:
Можно было бы сделать так, чтобы словари, необходимые для транскрипции и создания комплементарной цепи были бы атрибутами класса, а не атрибутами отдельных экземпляров. А то каждый раз при создании нового объекта класса у него будут в атрибуты записываться эти словари. 
```suggestion
class DNASequence(NucleicAcidSequence):
        comlementation = {'A': 'T',
            'T': 'A',
            'G': 'C',
            'C': 'G',
            'a': 't',
            't': 'a',
            'g': 'c',
            'c': 'g'
        }
         transcription = {
            'A': 'A',
            'T': 'U',
            'G': 'G',
            'C': 'C',
            'a': 'a',
            't': 'u',
            'g': 'g',
            'c': 'c'
        }
```